.github/workflows/release.yml

```diff
@@ -0,0 +1,47 @@
+name: Release Docker image
+
+on:
+  release:
+    types: [published]
+  pull_request:
+    types: [labeled, synchronize, reopened, ready_for_review, opened]
+
+env:
+  REGISTRY: ghcr.io
+  IMAGE_NAME: ${{ github.repository }}
+
+jobs:
+  build-and-push-image:
+    runs-on: ubuntu-latest
+    permissions:
+      contents: read
+      packages: write
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v3
+
+      - name: Log in to the Container registry
+        uses: docker/login-action@f054a8b539a109f9f41c372932f1ae047eff08c9
+        with:
+          registry: ${{ env.REGISTRY }}
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Extract metadata (tags, labels) for Docker
+        id: meta
+        uses: docker/metadata-action@98669ae865ea3cffbcbaa878cf57c20bbf1c6c38
+        with:
+          images: ${{ env.REGISTRY }}/${{ env.IMAGE_NAME }}
+
+      - name: Build and push Docker image
+        uses: docker/build-push-action@ad44023a93711e3deb337508980b4b5e9bcdc5dc
+        with:
+          context: .
+          build-args: |
+            GIT_COMMIT_SHA=${{ github.sha }}
+            GIT_REPOSITORY_URL=https://github.com/${{ github.repository }}
+            VERSION=${{ github.ref_name }}
+          push: true
```

Comment:
What we want is to only build the image, but not push it, if we are in a PR. So we only push releases.

But this could probably a 3rd workflow that runs on pushes on PRs (not `pull_request` events), with `push: false`

Reply:
If it only runs on `push` events to the pull request, won't that mean it won't run on creation of a pull request? Is that what we want?